lib/reconcile/utils.go

```diff
@@ -580,3 +661,55 @@ func getBaseAstarteAPIURL(cr *apiv1alpha1.Astarte) string {
 
 	return fmt.Sprintf("%s://%s", scheme, cr.Spec.API.Host)
 }
+
+func handleGenericUserCredentialsSecret(cr *apiv1alpha1.Astarte, c client.Client, scheme *runtime.Scheme, username, password, usernameKey, passwordKey, secretName string) error {
```

Comment:
For the sake of consistency, these functions keep `cr *apiv1alpha1.Astarte, c client.Client, scheme *runtime.Scheme` as the final arguments rather than the initial arguments.